tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -15,41 +15,335 @@
+                    "TEST_DAG_RUN_ID7",
+                    "TEST_DAG_RUN_ID8",
+                    "TEST_DAG_RUN_ID9",
+                    "TEST_DAG_RUN_ID10",
+                ],
+            ),
+            (
+                "api/v1/dags/TEST_DAG_ID/dagRuns?offset=0",
+                [
+                    "TEST_DAG_RUN_ID1",
+                    "TEST_DAG_RUN_ID2",
+                    "TEST_DAG_RUN_ID3",
+                    "TEST_DAG_RUN_ID4",
+                    "TEST_DAG_RUN_ID5",
+                    "TEST_DAG_RUN_ID6",
+                    "TEST_DAG_RUN_ID7",
+                    "TEST_DAG_RUN_ID8",
+                    "TEST_DAG_RUN_ID9",
+                    "TEST_DAG_RUN_ID10",
+                ],
+            ),
+            ("api/v1/dags/TEST_DAG_ID/dagRuns?limit=1&offset=5", ["TEST_DAG_RUN_ID6"]),
+            ("api/v1/dags/TEST_DAG_ID/dagRuns?limit=1&offset=1", ["TEST_DAG_RUN_ID2"]),
+            ("api/v1/dags/TEST_DAG_ID/dagRuns?limit=2&offset=2", ["TEST_DAG_RUN_ID3", "TEST_DAG_RUN_ID4"],),
+        ]
+    )
+    @provide_session
+    def test_handle_limit_and_offset(self, url, expected_dag_run_ids, session):
+        dagrun_models = self._create_dag_runs(10)
+        session.add_all(dagrun_models)
+        session.commit()
+
+        response = self.client.get(url)
+        assert response.status_code == 200
+
+        self.assertEqual(response.json["total_entries"], 10)
+        dag_run_ids = [dag_run["dag_run_id"] for dag_run in response.json["dag_runs"]]
+        self.assertEqual(dag_run_ids, expected_dag_run_ids)
+
+    @provide_session
+    def test_should_respect_page_size_limit(self, session):
+        dagrun_models = self._create_dag_runs(200)
+        session.add_all(dagrun_models)
+        session.commit()
+
+        response = self.client.get("api/v1/dags/TEST_DAG_ID/dagRuns?limit=150")
+        assert response.status_code == 200
+
+        self.assertEqual(response.json["total_entries"], 200)
+        self.assertEqual(len(response.json["dag_runs"]), 100)
+
+    def _create_dag_runs(self, count):
+        return [
+            DagRun(
+                dag_id="TEST_DAG_ID",
+                run_id="TEST_DAG_RUN_ID" + str(i),
+                run_type=DagRunType.MANUAL.value,
+                execution_date=timezone.parse(self.default_time) + timedelta(minutes=i),
+                start_date=timezone.parse(self.default_time),
+                external_trigger=True,
+            )
+            for i in range(1, count + 1)
+        ]
+
+
+class TestGetDagRunsPaginationFilters(TestDagRunEndpoint):
+    @parameterized.expand(
+        [
+            (
+                "api/v1/dags/TEST_DAG_ID/dagRuns?start_date_gte=2020-06-18T18:00:00+00:00",
+                ["TEST_START_EXEC_DAY_18", "TEST_START_EXEC_DAY_19"],
+            ),
+            (
+                "api/v1/dags/TEST_DAG_ID/dagRuns?start_date_lte=2020-06-11T18:00:00+00:00",
+                ["TEST_START_EXEC_DAY_10"],
```

Comment:
```suggestion
                ["TEST_START_EXEC_DAY_10", "TEST_START_EXEC_DAY_11"],
```
We have an inclusive filter, so it should probably be included as well. However, this may change in the future.

Related issue: 
https://github.com/apache/airflow/issues/9237

Reply:
This seems like a problem with `<=` operator in sqlalchemy. It returns only the less than values. 
I have also tried separating the operator and using the `or_` operator but it's still not inclusive
I am wondering if there is another way I could do it?

Reply:
I have a day off today, so I have limited options to check it out, but it can help you.
https://github.com/apache/airflow/blob/master/TESTING.rst#tracking-sql-statements

Reply:
Thanks for the link, it helped. 
What was happening was that on the view dates that have this format '2020-06-11T18:00:00+00:00' are changed to '2020-06-11T18:00:00 00:00'. That is, the + are replaced with empty space. So when it is parsed, only the date part were parsed. I then replaced +00:00 with Z and the problem was solved.

Maybe we should open an issue for custom connexion datetime format validator

Reply:
The fix https://github.com/apache/airflow/pull/9153/commits/7a72935b63bdb6c5622f8c00d4a3d2f996eaa2ae

Reply:
This is the command I used to investigate:
`pytest --trace-sql=num,sql,parameters --capture=no   tests/api_connexion/endpoints/test_dag_run_endpoint.py -k test_date_filters_gte_and_lte_1_api_v1_dags_TEST_DAG_ID_dagRuns_start_date_lte_2020_06_11T18_00_00_00_00`

I set pdb debuger at this point in dag_run_endpoint:

`    if start_date_lte:
        import pdb; pdb.set_trace()
        query = query.filter(DagRun.start_date <= timezone.parse(start_date_lte))`

Check image below:

![start_date_lte](https://user-images.githubusercontent.com/4122866/84494081-c484be00-aca0-11ea-9ec6-4a717cf1d15e.png)